### ISSUE TITLE
Nested grid layout not correct

### ISSUE THREAD

**Issue by scastria** [xamarin/Xamarin.Forms#4578]
### Description
There is something wrong with a parent Grid layout calculations when there is a child Grid.  Attached is a sample program that shows the parent Grid correctly expands to contain its child grid vertically, but does NOT do it horizontally.

### Steps to Reproduce

1. Run attached program
2. Click Vertical to see proper behavior
3. Click Horizontal to see BAD behavior

### Expected Behavior

Horizontal behavior should show parent Grid expand to contain its child grid

### Actual Behavior

Child grid gets drawn outside the bounds of the parent grid.

### Basic Information

- Version with issue:
- Last known good version:
- IDE:  VS Mac
- Platform Target Frameworks: 
  - iOS:  12.2.1.10
  - Android: 9.0.0.20

### Screenshots
![simulator screen shot - iphone 8 - 2018-11-28 at 22 04 40](https://user-images.githubusercontent.com/4534494/49200529-74623700-f35a-11e8-999b-4ba2742d28dc.png)
![simulator screen shot - iphone 8 - 2018-11-28 at 22 04 36](https://user-images.githubusercontent.com/4534494/49200535-77f5be00-f35a-11e8-834e-436ab33e0c53.png)

### Reproduction Link

[TestPopup.zip](https://github.com/xamarin/Xamarin.Forms/files/2627664/TestPopup.zip)

**Comment by samhouts** [comment]
Looks like it expands if you set your column definitions to `Width = GridLength.Auto` instead of `GridLength.Star`. Do you see that, too?

**Comment by scastria** [comment]
@samhouts Yes.  If you feel it is working correctly, you can close the issue.  I will keep playing with it.  Thanks.
